include/mbedtls/config_psa.h

```diff
@@ -220,6 +220,65 @@ extern "C" {
 #endif /* !MBEDTLS_PSA_ACCEL_KEY_TYPE_RSA_PUBLIC_KEY */
 #endif /* PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY */
 
+#if defined(PSA_WANT_KEY_TYPE_AES)
+#if !defined(MBEDTLS_PSA_ACCEL_KEY_TYPE_AES)
+#define MBEDTLS_PSA_BUILTIN_KEY_TYPE_AES 1
+#define MBEDTLS_AES_C
+#endif /* !MBEDTLS_PSA_ACCEL_KEY_TYPE_AES */
+#endif /* PSA_WANT_KEY_TYPE_AES */
+
+#if defined(PSA_WANT_KEY_TYPE_ARC4)
+#if !defined(MBEDTLS_PSA_ACCEL_KEY_TYPE_ARC4)
+#define MBEDTLS_PSA_BUILTIN_KEY_TYPE_ARC4 1
+#define MBEDTLS_ARC4_C
+#endif /*!MBEDTLS_PSA_ACCEL_KEY_TYPE_ARC4 */
+#endif /* PSA_WANT_KEY_TYPE_ARC4 */
+
+#if defined(PSA_WANT_KEY_TYPE_CAMELLIA)
+#if !defined(MBEDTLS_PSA_ACCEL_KEY_TYPE_CAMELLIA)
+#define MBEDTLS_PSA_BUILTIN_KEY_TYPE_CAMELLIA 1
+#define MBEDTLS_CAMELLIA_C
+#endif /*!MBEDTLS_PSA_ACCEL_KEY_TYPE_CAMELLIA */
+#endif /* PSA_WANT_KEY_TYPE_CAMELLIA */
+
+#if defined(PSA_WANT_KEY_TYPE_DES)
+#if !defined(MBEDTLS_PSA_ACCEL_KEY_TYPE_DES)
+#define MBEDTLS_PSA_BUILTIN_KEY_TYPE_DES 1
+#define MBEDTLS_DES_C
+#endif /*!MBEDTLS_PSA_ACCEL_KEY_TYPE_DES */
+#endif /* PSA_WANT_KEY_TYPE_DES */
+
+#if defined(PSA_WANT_ALG_STREAM_CIPHER)
+    /* Nothing to define */
+#endif /* PSA_WANT_ALG_STREAM_CIPHER */
+
+#if defined(PSA_WANT_ALG_CTR)
```

Comment:
Why no possibility of acceleration for modes? It may be ok, just want to check this is not an oversight.

Reply:
I wasn't sure quite how this would work. I guess I'd normally expect the acceleration to be kind of a homogeneous thing, that a combination of key type and mode would be accelerated together. Should I perhaps define accelerations for those here? I'm not sure what happens if a mode acceleration is defined, but not a key type (although, I guess defining those properly is up to the acceleration implementation.

Reply:
Let's focus on the simple case, which is the homogeneous case: a driver can be used for a keytype+mode combination only if it advertises both. For example, AES-CTR should route to a driver only if `MBEDTLS_PSA_ACCEL_KEY_TYPE_AES && MBEDTLS_PSA_ACCEL_ALG_CTR`. Software AES should be omitted only if all wanted modes have a driver.

In stage 2 we'll implement leveraging accelerated ECB to implement other modes in software.

In stages 3 and beyond we can worry about more exotic cases, such as leveraging GHASH hardware with a software implementation of CAMELLIA, or leveraging an AES-CTR driver for a hardware/software hybrid AES-GCM.